draft-js-inline-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -52,16 +39,32 @@ export default class Toolbar extends React.Component {
     // when focusing editor with already present selection
     setTimeout(() => {
       if (!this.toolbar) return;
-      const relativeParent = getRelativeParent(this.toolbar.parentElement);
-      const toolbarHeight = this.toolbar.clientHeight;
-      const relativeRect = (relativeParent || document.body).getBoundingClientRect();
-      const selectionRect = getVisibleSelectionRect(window);
 
+      // The editor root should be two levels above the node from
+      // `getEditorRef`. In case this changes in the future, we
+      // attempt to find the node dynamically by traversing upwards.
+      const editorRef = this.props.store.getItem('getEditorRef')();
```

Comment:
we already a method (above) for finding the relative parent. will that not do the job?

Reply:
We're not looking for the next enclosing positioned parent, but rather for the DOM node with the class `DraftEditor-root` – so I think we don't need it anymore. Btw. that method does the same as simply calling `node.offsetParent` 🙂

Reply:
before doing any more work on this, mind having a look at https://github.com/draft-js-plugins/draft-js-plugins/pull/1018, there's some good stuff in there. We have two prs now that fix this problem 😨

Reply:
That's an interesting PR! I think it's doing a different thing though. That PR adds support for a smart positioning of the inline toolbar and its caret. As far as I can tell the positioning within a relative parent isn't addressed there. Maybe we can merge this PR and rebase the other PR on top of this to use the relevant parts?